lib/json/common.rb

```diff
@@ -589,21 +589,34 @@ class << self
     :allow_nan   => true,
   }
 
-  # Dumps _obj_ as a JSON string, i.e. calls generate on the object and returns
-  # the result.
+  # :call-seq:
+  #   JSON.dump(obj, io = nil, limit = nil)     
+  #
+  # Constructs a \JSON \String from +obj+ by calling
+  #   JSON.generate(obj, JSON.dump_default_options.merge(:max_nesting => limit))
```

Comment:
How about translating this to words? Note: IIRC, `JSON` is also a module that can be included, so the call sequence isn't necessarily `JSON.dump`, and it calls `generate`, which may or may not be `JSON.generate` depending on how dump is called, right?

Reply:
Suggested words, please?

Module/class issue is a serious oversight on my part.  The only two example calls in the existing doc use JSON.parse and JSON.generate, and I followed that convention.

This is the last method in JSON that I'm 'enhancing,' and it, like the others, uses the class methods in its examples.

The doc for all the methods should be revised to take the 'duality' into account;  the introductory material should also address this.

I suggest we merge with the calls as is for now, and figure out a good way to do better, afterwards.

Reply:
I've restored the original first sentence.